docs/collections.md

```diff
@@ -0,0 +1,271 @@
+2. Create the necessary tasks and add them to the collection.
+  - Pass variables calculated in the first phase to task parameters.
+3. Run the tasks via `$collection->run()`.
+  - Check and report errors once after `run()` returns.
+
+Following this pattern will keep your code linear and easy to understand.
+
+## Basic Collection Example
+
+A simple example is shown below:
+
+``` php
+<?php
+class RoboFile extends \Robo\Tasks
+{
+    function myOperation()
+    {
+        // Determine the value of any variables that will control
+        // task execution.
+        $baseDir = 'logs';
+        
+        // Create a collection.
+        $collection = $this->collection();
+        
+        // Add some filesystem tasks to the collection
+        $this->taskFileSystemStack()
+          ->mkdir($baseDir)
+          ->touch("$baseDir/.gitignore")
+          ->chgrp($baseDir, 'www-data')
+          ->symlink('/var/log/nginx/error.log', '$baseDir/error.log')
+          ->addToCollection($collection);
+        
+        // Add some other tasks.
+        $this->taskOther($baseDir)
+          ->addToCollection($collection);
+        
+        // Run all of our tasks.
+        $result = $collection->run();
+    }
+}
+?>
+```
+
+When `$collection->run()` is executed, all of the tasks in the collection will run, and the final result (a Robo\Result object) will be returned. If any of the tasks fail, then the tasks that follow are skipped.
+
+## Rollbacks
+
+It is also possible to add rollback tasks to a collection.  A rollback task is executed if all of the tasks that come before it succeed, and at least one of the tasks that come after it fails.  If all tasks succeed, then no rollback tasks are executed.
+
+### Rollback Tasks
+
+Rollback tasks can be used to clean up after failures, so the state of the system does not change when execution is interrupted by an error. An example of this is shown below:
+
+``` php
+<?php
+class RoboFile extends \Robo\Tasks
+{
+    function myOperation()
+    {
+        $collection = $this->collection();
+        $this->taskFileSystemStack()
+          ->mkdir('work')
+          ->addToCollection($collection);
+        
+        // Add a rollback task
+        $this->taskFileSystemStack()
+          ->rmdir('work')
+          ->addAsRollback();
+
+        $this->taskOther('work')
+          ->addToCollection($collection);
+        
+        $result = $collection->run();
+    }
+}
+?>
+```
+
+In the example above, if the `mkdir('work')` fails, then neither the rollback task nor the 'taskOther' task will be executed.  If it works, though, then `taskOther()` will also be executed; if `taskOther()` fails, then the rollback task will be run, and the 'work' directory will be removed.
+
+### Rollback Methods
+
+Any task may also implement \Robo\Contract\RollbackInterface; if this is done, then its `rollback()` method will be called if the task is `run()` on a collection that later fails.
+
+## Completions
```

Comment:
I'm not opposed to renaming, but a bit worried about confusion if we get too close to the reserved words. Also, completions do not fire if someone throws an exception that is not caught, so avoiding exception terminology might be best here.